Src/ILGPU/Runtime/Cuda/CudaAccelerator.cs

```diff
@@ -482,6 +482,24 @@ public long GetFreeMemory()
             where T : unmanaged =>
             Allocate2DPitchedY<T>(extent, PitchedAllocationAlignmentInBytes);
 
+        /// <summary>
+        /// Maps a buffer allocated by another process of the specified size in bytes on this accelerator.
+        /// </summary>
+        /// <param name="ipcMemHandle">A IPC memory handle from another process.</param>
+        /// <param name="length">The number of elements to allocate.</param>
+        /// <param name="elementSize">The size of a single element in bytes.</param>
+        /// <returns>A mapped buffer of shared memory on this accelerator.</returns>
+        public CudaIpcMemoryBuffer MapRaw(CudaIpcMemHandle ipcMemHandle, long length, int elementSize)
```

Comment:
I'm thinking this method signature should be something like:
`public MemoryBuffer MapFromIpcMemHandle(CudaIpcMemHandle ipcMemHandle, long length, int elementSize)`.

We do not need to expose the underlying `CudaIpcMemoryBuffer`.

And we should also have another method like:
`public MemoryBuffer1D<T, Stride1D.Dense> MapFromIpcMemHandle1D<T>(CudaIpcMemHandle ipcMemHandle, long length)`.

This is the convenience method that will make it easier to use. Not sure if we need the 2D or 3D variants.

I'm also wondering if we should be using the `Map` prefix, or if it should be an 'Allocate' prefix, so that it is discoverable with the rest of the memory allocation functions when using auto-complete. e.g. `AllocateFromIpcMemHandle`.

Reply:
Agree on returning `MemoryBuffer` and naming the method `MapFromIpcMemHandle`.
My original idea with `MapRaw` was to mirror `Accelerator.AllocateRaw`, but `MapFrom` works as well.
I also did think about using `Allocate`, but decided against it, because it is exactly what we're **not** doing. The memory was already allocated by a different process, what we're creating is more like a `View`. Instead I went with `Map` because that is also what they use in the [documentation](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__DEVICE.html#group__CUDART__DEVICE_1g01050a29fefde385b1042081ada4cde9) ("Maps memory exported from another process").

As for having a similar API to `Allocate1D()` like e.g. `Map1D`, that would be something I could do. As far I understand the code, I most likely can just replace `AllocateRaw` with its IPC equivalent and move the `CudaIpcMemHandle` through the methods. The methods then should be in `CudaAccelerator` as opposed to `Accelerator`.